lib/recipes-data/src/lib/extract-recipes.ts

```diff
@@ -0,0 +1,37 @@
+import type {Block} from "@guardian/content-api-models/v1/block";
+import type {Blocks} from "@guardian/content-api-models/v1/blocks";
+import type {Content} from "@guardian/content-api-models/v1/content";
+import {ContentType} from "@guardian/content-api-models/v1/contentType";
+import {ElementType} from "@guardian/content-api-models/v1/elementType";
+import type {RecipeReferenceWithoutChecksum} from './models';
+
+export function extractAllRecipesFromArticle(content: Content): Array<RecipeReferenceWithoutChecksum | null> {
+  if (content.type == ContentType.ARTICLE && content.blocks) {
+    const articleBlocks: Blocks = content.blocks
+    const getAllMainBlockRecipesIfPresent: Array<RecipeReferenceWithoutChecksum | null> = extractRecipeData(content.id, articleBlocks.main as Block)
+    const bodyBlocks = articleBlocks.body as Block[]
+    const getAllBodyBlocksRecipesIfPresent: Array<RecipeReferenceWithoutChecksum | null> = bodyBlocks.flatMap(bodyBlock => extractRecipeData(content.id, bodyBlock))
+    return getAllMainBlockRecipesIfPresent.concat(getAllBodyBlocksRecipesIfPresent)
+  } else {
+    return Array<RecipeReferenceWithoutChecksum>()
+  }
+}
+
+export function extractRecipeData(canonicalId: string, block: Block): Array<RecipeReferenceWithoutChecksum | null> {
+  return block.elements
+    .filter(elem => elem.type === ElementType.RECIPE)
+    .map(recp => parseJsonBlob(canonicalId, recp.recipeTypeData?.recipeJson as string))
+    .filter(recp => !!recp)
```

Comment:
if this is here, you can set the return type to `RecipeReferenceWithoutChecksum[]` as there is no way it's going to return nulls.

If TS complains about that, you're safe to put `as RecipeReferenceWithoutChecksum[]` onto the end of line 24 to tell it there's no problem

Reply:
Yup, TS complained as you predicted, So had to return with `as` in it..